### ISSUE TITLE
Plugin fails to read big-endian PCAP

### ISSUE THREAD

**Issue by erik4711** [emnahum/zeek-pcapovertcp-plugin#2]
The PcapOverTcp plugin fails to read big-endian capture files (starting with `a1 b2 c3 d4`). Zeek prints the following error message when big-endian data is read using `-i pcapovertcp::IP:PORT`:
```
error: Attempted to unregister an unknown file descriptor 18 from PktSrc
fatal error: problem with interface pcapovertcp::127.0.0.1:57013 (cannot compile BPF filter "": unknown data link type -50331648 (min 104, max 289))
```
Zeek has no problems reading the same big-endian traffic from a file with `-r`.

An example capture file in big-endian PCAP format can be downloaded here:
https://www.netresec.com/files/proxy-191023-091924.pcap

**Comment by emnahum** [comment]
Hmm.  That's correct.  I could put in a check after reading the magic number, and simply bail out if the magic number is d4c3b2a1, saying the code doesn't (yet) support big-endian, or I could add the support.  Two questions:

- Are big endian machines very common?  As far as I can see, most are little-endian or at least bi-endian.  Most big-endian machines are rare/extinct.  Are you simply testing a big endian PCAP, or do you have a real use case?

- Is there any code that illustrates how to do this other than the libpcap source?  My recollection is that it's complex.

Basically I'm trying to figure out the cost/benefit of adding support.

**Comment by erik4711** [comment]
Big endian PCAP files are not very common, but you typically see them when capturing traffic on a firewall, switch or other type of network device. Here's one such example (with a big-endian capture file available for download): https://mrncciew.com/2022/08/30/9800-packet-captures/

It's not very complicated to implement big-endian support. You just need to reverse the byte order for each field in the 24 byte PCAP header and 16 byte packet header.  All fields are 4 bytes (int32), except for major and minor version, which are 2 bytes (short).

**Comment by emnahum** [comment]
You are correct, I checked in the libpcap source and it is very straightforward.  I've made the changes and pushed the latest version.  It should appear in the packages repo within a day.
